tools/infrasys/src/error.rs

```diff
@@ -0,0 +1,160 @@
+    ))]
+    ParseResponse { what: String, resource_name: String },
+
+    #[snafu(display("Failed to get bucket policy statement for bucket {}", bucket_name))]
+    GetPolicyStatement { bucket_name: String },
+
+    #[snafu(display(
+        "Failed to update bucket policy for bucket {}: {}",
+        bucket_name,
+        source
+    ))]
+    PutPolicy {
+        bucket_name: String,
+        source: rusoto_core::RusotoError<rusoto_s3::PutBucketPolicyError>,
+    },
+
+    #[snafu(display("Failed to push object to bucket {}: {}", bucket_name, source))]
+    PutObject {
+        bucket_name: String,
+        source: rusoto_core::RusotoError<rusoto_s3::PutObjectError>,
+    },
+
+    #[snafu(display("Missing environment variable '{}'", var))]
+    Environment {
+        var: String,
+        source: std::env::VarError,
+    },
+
+    #[snafu(display("Failed to read file at {}: {}", path.display(), source))]
+    FileRead { path: PathBuf, source: io::Error },
+
+    #[snafu(display("Failed to open file at {}: {}", path.display(), source))]
+    FileOpen { path: PathBuf, source: io::Error },
+
+    #[snafu(display("Failed to write file in {}: {}", path.display(), source))]
+    FileWrite { path: PathBuf, source: io::Error },
+
+    #[snafu(display("File already exists at {}", path.display()))]
+    FileExists { path: PathBuf },
+
+    #[snafu(display(
+        "Failed to create keys due to invalid key config. Missing {}.",
+        missing
+    ))]
+    KeyConfig { missing: String },
+
+    #[snafu(display(
+        "Failed to create new keys or access pre-existing keys in available_keys list."
+    ))]
+    KeyCreation,
+
+    #[snafu(display("Invalid path {} for {}", path.display(), thing))]
+    Path { path: PathBuf, thing: String },
+
+    #[snafu(display("Failed to create directory {}: {}", path.display(), source))]
+    Mkdir { path: PathBuf, source: io::Error },
+
+    #[snafu(display("Failed to convert {} to URL: {}", input, source))]
+    ParseUrl {
+        input: String,
+        source: url::ParseError,
+    },
+
+    #[snafu(display("Failed to parse {} to int: {}", what, source))]
+    ParseInt {
+        what: String,
+        source: std::num::ParseIntError,
+    },
+
+    #[snafu(display("Publication/Root key threshold must be <= {}, currently {}", num_keys.to_string(), threshold))]
+    InvalidThreshold { threshold: String, num_keys: usize },
+
+    #[snafu(display("Failed to convert updated Infra.toml information to yaml: {}", source))]
+    InvalidYaml { source: serde_yaml::Error },
+
+    #[snafu(display("Failed to convert {} to yaml: {}", what, source))]
+    InvalidJson {
+        what: String,
+        source: serde_json::Error,
+    },
+
+    #[snafu(display("Failed to get parent of path: {}", path.display()))]
+    Parent { path: PathBuf },
+
+    #[snafu(display("Logger setup error: {}", source))]
+    Logger { source: log::SetLoggerError },
+
+    #[snafu(display("Returned {}: tuftool {}", code, command))]
```

Comment:
I didn't notice that we were using tuftool. I had hoped we could use `tough` as a library instead of shelling out to the CLI.

Reply:
Ah yes, we would need to move a few things from tuftool --> tough, which might take a little longer than the length of my internship :(
But adding this to the next steps. We can also refactor pubsys tuftool use while we're at it!

Reply:
As we discussed, using `tough` instead of `tuftool` will have to be out of scope for now since we don't have the cycles to make `tough` do everything we need.

Reply:
> I'd reword this as
> 
> ```
> ("tuftool {} returned {}", command, code)
> ```
> 
> Also, is there any more context we can give the user in this case?

Because we don't know where tuftool failed, I think the only thing we can provide them is the error code? Doesn't look like ExitStatus outputs much else: https://doc.rust-lang.org/std/process/struct.ExitStatus.html 
But I might just be missing a way to grab more info -- can definitely add it in if there's a way